src/Security/src/Microsoft.AspNetCore.Authentication.OpenIdConnect/ClientAuthentication/BasicAuthentication.cs

```diff
@@ -0,0 +1,46 @@
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace Microsoft.AspNetCore.Authentication.OpenIdConnect
+{
+    /// <summary>
+    /// Use basic authorization header
+    /// </summary>
+    public class BasicAuthentication : IClientAuthentication
+    {
+        private readonly bool withEscaping;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="withEscaping">allow to do EscapeData on the clientId and clientSecret</param>
+        public BasicAuthentication(bool withEscaping = true)
+        {
+            this.withEscaping = withEscaping;
```

Comment:
Why would you ever turn this off?

Reply:
because some openid server dont correctly implement this and not unescape. Look at @PinpointTownes  comment's